src/app.js

```diff
@@ -1,21 +1,53 @@
 'use strict';
+require('dotenv').config();
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
-
-module.exports = sum;
+const path = require('path');
+const fs = require('fs');
+const http = require('http');
+const { pipeline } = require('stream');
+const { getContentType } = require('./utils/getContentType');
+const { sendResError } = require('./utils/sendResError');
+
+const PORT = process.env.PORT || 3000;
+
+const server = http.createServer((req, res) => {
+  const [pathNameFile, ...pathNames] = req.url.split('/').filter(i => i);
+  let pathResponseFile;
+
+  if (!pathNameFile || pathNameFile !== 'file') {
+    sendResError(res, 'url string should contain \'file\'');
+
+    return;
+  }
+
+  if (pathNames.length === 0) {
+    pathResponseFile = path.join('public', 'index.html');
+  } else {
+    pathResponseFile = path.join('public', ...pathNames);
+  }
+
+  if (!fs.existsSync(pathResponseFile)) {
+    sendResError(res, 'file not exists');
+
+    return;
+  }
+
+  const file = fs.createReadStream(pathResponseFile);
+  const extentionFile = pathResponseFile.split('.').pop();
+
+  res.setHeader('Content-Type', getContentType(extentionFile));
+
+  pipeline(file, res, () => {
```

Comment:
The pipeline callback function is incorrect. It should have parameters (err) and handle the error if any. For example: pipeline(file, res, (err) => { if (err) { sendResError(res, 'something went wrong', 500); } }).